File: Python Word Examples/ShapesApiDemo/updateShapeFormat.py
import spirecloudword
from spirecloudword import ShapeFormat, Color
from spirecloudword.configuration import *

appId = "your id"
appKey = "your key"
baseUrl = "https://api.e-iceblue.cn"
configuration = Configuration(appId, appKey,baseUrl)
api= spirecloudword.api.ShapesApi(configuration)

file_name = 'Shape.docx'
paragraph_path = 'Section/0/Body/0/Paragraph/0'
index = 0
shape_properties = ShapeFormat(50, 50, 'Octagon')
shape_properties._horizontal_origin = 'LeftMarginArea'
shape_properties._vertical_origin = 'Paragraph'
shape_properties._vertical_position = 80
shape_properties._horizontal_position = 80
shape_properties._fill_color = Color(100, 100, 100)
shape_properties._rotation = 45
shape_properties._stroke_weight = 2
shape_properties._stroke_color = Color(0, 206, 209)
shape_properties._text_wrapping_type = 'Both'
shape_properties._text_wrapping_style = 'InFrontOfText'
shape_properties._z_order = 1
extra = {'folder':"input", 'dest_file_path': 'output/UpdateShapeFormat.docx'}
result = api.update_shape_format(file_name, paragraph_path, index, shape_properties, **extra)

File: JavaScript Word Examples/ShapesApiDemo/updateShapeFormat.js
(function (){
    var appId = "your id";
    var appKey = "your key";    
    var baseUrl = "https://api.e-iceblue.cn";

    var Spirecloudword = require('../../src/index');
    var apiClient = new Spirecloudword.ApiClient(appId, appKey,baseUrl);
    var instance = new Spirecloudword.ShapesApi(apiClient);

    var fileName = "Shape.docx";
    var paragraphPath = "Section/0/Body/0/Paragraph/0";
    var index = 0;
    var tempVar = new Spirecloudword.ShapeFormat();
    tempVar.width = 50;
    tempVar.height = 50;
    tempVar.shapeType = "Octagon";
    tempVar.horizontalOrigin = "LeftMarginArea";
    tempVar.verticalOrigin = "Paragraph";
    tempVar.verticalPosition = 80.0;
    tempVar.horizontalPosition = 80.0;
    tempVar.fillColor = new Spirecloudword.Color();
    tempVar.fillColor.red = 100;
    tempVar.fillColor.green = 100;
    tempVar.fillColor.blue = 100;
    tempVar.rotation = 0;
    tempVar.strokeWeight = 2;
    tempVar.strokeColor = new Spirecloudword.Color();
    tempVar.strokeColor.red = 0;
    tempVar.strokeColor.green = 206;
    tempVar.strokeColor.blue = 209;
    tempVar.textWrappingType = "Both";
    tempVar.textWrappingStyle = "InFrontOfText";
    tempVar.zOrder = 1;
    var shapeProperties = tempVar;
    var opts = { folder: 'input' };
    var destFilePath = "output/UpdateShapeFormat_out.docx";
    instance.updateShapeFormat(fileName, paragraphPath, index, shapeProperties, destFilePath, opts,
        function (error, data, response) {
            if (error) throw error;
        });
})();